src/main/scala/edu/colorado/plv/cuanto/abstracting/tc/domains/Interval.scala

```diff
@@ -0,0 +1,111 @@
+package edu.colorado.plv.cuanto
+package abstracting.tc
+package domains
+
+/** A domain representing an integer constrained by an upper and/or
+  * lower bound
+  *
+  * Elements of the [[Interval]] domain should be constructed using
+  * the [[btw `btw`]], [[lte `lte`]], and [[gte `gte`]] operations
+  * defined in the companion object.
+  *
+  * @author octalsrc
+  */
+package object Interval {
+  import math.{min,max}
+
+  sealed abstract class Interval
+
+  private[this] case object Top extends Interval
+  private[this] case object Bot extends Interval
+
+  private[this] case class Gte(gte: Int) extends Interval
+  private[this] case class Lte(lte: Int) extends Interval
+  private[this] case class Btw(gte: Int, lte: Int) extends Interval
+
+  private[this] type I = Interval
+
+  private[this] val reduce: I => I = e => e match {
+    case Btw(g,l) if g > l => Bot
+    case e => e
+  }
```

Comment:
It's just a utility function for the `btw` smart constructor; I'll move it to `btw`'s definition